src/test/java/org/alfresco/repo/copy/CopyServiceImplUnitTest.java

```diff
@@ -0,0 +1,157 @@
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package org.alfresco.repo.copy;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Locale;
+
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.alfresco.service.cmr.repository.StoreRef;
+import org.alfresco.service.namespace.QName;
+import org.alfresco.util.GUID;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.extensions.surf.util.I18NUtil;
+
+/**
+ * Unit tests for {@link CopyServiceImpl} class.
+ *
+ * @author Sara Aspery
+ */
+public class CopyServiceImplUnitTest
+{
+    private static final String FILE_NAME = "Test File";
+    private static final String FILE_EXTENSION = ".txt";
+    private static final QName ASSOC_TYPE_QNAME = QName.createQName("assoc://type/");
+
+    /* I18N labels used by the tests */
+    private static final String COPY_OF_LABEL = "copy_service.copy_of_label";
+
+    private CopyServiceImpl copyServiceImpl;
+
+    private Locale preservedLocale;
+    private String copyOfLabelTranslated;
+    private NodeRef destinationParent;
+
+    @Before
+    public void setup()
+    {
+        I18NUtil.registerResourceBundle("alfresco/messages/copy-service");
+        this.preservedLocale = I18NUtil.getLocale();
+        this.copyOfLabelTranslated = I18NUtil.getMessage(COPY_OF_LABEL, "");
+        this.destinationParent = new NodeRef(StoreRef.STORE_REF_WORKSPACE_SPACESSTORE, GUID.generate());
+
+        this.copyServiceImpl = new CopyServiceImpl();
+    }
+
+    @Test
+    public void testBuildNewName_FileWithExtension()
+    {
+        String[] expectedNamesOfCopies = generateCopyNames(FILE_NAME, FILE_EXTENSION,3,true);
+        testAndAssertMultipleFileCopies(expectedNamesOfCopies);
+    }
+
+    @Test
+    public void testBuildNewName_FileWithoutExtension()
+    {
+        String[] expectedNamesOfCopies = generateCopyNames(FILE_NAME, "",3,true);
+        testAndAssertMultipleFileCopies(expectedNamesOfCopies);
+    }
+
+    @Test
+    public void testBuildNewName_FileWithExtension_JapaneseLocale()
+    {
+        switchLocale(Locale.JAPANESE);
+
+        String[] expectedNamesOfCopies = generateCopyNames(FILE_NAME, FILE_EXTENSION,3,false);
+        testAndAssertMultipleFileCopies(expectedNamesOfCopies);
+
+        restoreLocale();
+    }
+
+    @Test
+    public void testBuildNewName_FileWithoutExtension_JapaneseLocale()
+    {
+        switchLocale(Locale.JAPANESE);
+
+        String[] expectedNamesOfCopies = generateCopyNames(FILE_NAME, "",3,false);
+        testAndAssertMultipleFileCopies(expectedNamesOfCopies);
+
```

Comment:
The tests are not clear. A developer would have to work out what the code is doing or put in debug statements to understand how  the Copy Of should work. See examples tests below that make the behaviour of the buildNewName method more obvious.

Also for the tests that are not Japanese you should set the Locale to ENGLISH so that they would work for developers with a default Locale of Japanese.

    @Test
    public void testBuildNewName_FileWithExtension()
    {
        switchLocale(Locale.ENGLISH);
        assertEquals("Copy of filename.txt", copyServiceImpl.buildNewName("filename.txt"));
        restoreLocale();
    }

    @Test
    public void testBuildNewName_FileWithoutExtension()
    {
        switchLocale(Locale.ENGLISH);
        assertEquals("Copy of filename", copyServiceImpl.buildNewName("filename"));
        restoreLocale();
    }

    @Test
    public void testBuildNewName_FileWithExtensionAndCopyOf()
    {
        switchLocale(Locale.ENGLISH);
        assertEquals("Copy of Copy of filename.txt", copyServiceImpl.buildNewName("Copy of filename.txt"));
        restoreLocale();
    }
    
    @Test
    public void testBuildNewName_JapaneseWithExtension()
    {
        switchLocale(Locale.JAPANESE);
        assertEquals("filename のコピー.txt", copyServiceImpl.buildNewName("filename.txt"));
        restoreLocale();
    }

    @Test
    public void testBuildNewName_JapaneseWithoutExtension()
    {
        switchLocale(Locale.JAPANESE);
        assertEquals("filename のコピー", copyServiceImpl.buildNewName("filename"));
        restoreLocale();
    }

    @Test
    public void testBuildNewName_JapaneseWithExtensionAndCopyOf()
    {
        switchLocale(Locale.JAPANESE);
        assertEquals("filename のコピー のコピー.txt", copyServiceImpl.buildNewName("filename のコピー.txt"));
        restoreLocale();
    }